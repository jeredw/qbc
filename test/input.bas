input hello
input a, b, c
input ; foo$, bar%
input ; "y/n", thing
input "y/n"; yn
INPUT "Enter name and age: ";Person.FullName,Person.Age
input #1, foo$, bar%, baz$
input #a%, foo$, bar%, baz$
