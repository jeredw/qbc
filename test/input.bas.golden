(program 
  (statement 
    (input_statement input 
      (variable_or_function_call hello))) \n 
  (statement 
    (input_statement input 
      (variable_or_function_call a) , 
      (variable_or_function_call b) , 
      (variable_or_function_call c))) \n 
  (statement 
    (input_statement input ; 
      (variable_or_function_call foo$) , 
      (variable_or_function_call bar%))) \n 
  (statement 
    (input_statement input ; "y/n" , 
      (variable_or_function_call thing))) \n 
  (statement 
    (input_statement input "y/n" ; 
      (variable_or_function_call yn))) \n 
  (statement 
    (input_statement INPUT "Enter name and age: " ; 
      (variable_or_function_call Person.FullName) , 
      (variable_or_function_call Person.Age))) \n 
  (statement 
    (input_statement input 
      (file_number # 
        (expr 
          (literal 1))) , 
      (variable_or_function_call foo$) , 
      (variable_or_function_call bar%) , 
      (variable_or_function_call baz$))) \n 
  (statement 
    (input_statement input 
      (file_number # 
        (expr 
          (variable_or_function_call a%))) , 
      (variable_or_function_call foo$) , 
      (variable_or_function_call bar%) , 
      (variable_or_function_call baz$))) \n)
